pkg/operator/controllers/azurensg/subnetnsg_test.go

```diff
@@ -0,0 +1,71 @@
+package azurensg
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"strings"
+	"testing"
+
+	mgmtnetwork "github.com/Azure/azure-sdk-for-go/services/network/mgmt/2019-07-01/network"
+	"github.com/Azure/go-autorest/autorest/azure"
+	"github.com/Azure/go-autorest/autorest/to"
+	"github.com/golang/mock/gomock"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	utillog "github.com/Azure/ARO-RP/pkg/util/log"
+	mock_network "github.com/Azure/ARO-RP/pkg/util/mocks/azureclient/mgmt/network"
+)
+
+var (
+	vnet = azure.Resource{
+		SubscriptionID: "0000000-0000-0000-0000-000000000000",
+		ResourceGroup:  "vnet-rg",
+		ResourceName:   "vnet-name",
+	}
+)
+
+func TestFixSubnet(t *testing.T) {
+	r := AzureNSGReconciler{log: utillog.GetLogger()}
+	for _, tt := range []struct {
+		nsgname             string
+		architectureVersion api.ArchitectureVersion
+		expectedValue       string
+	}{
```

Comment:
Please add `name string` and a name describing each test case and wrap each run with `t.Run( tt.name, ...)` (see an example [here](https://github.com/Azure/ARO-RP/blob/7d5ee827c963740c63563f4a3a9de027ef43a9b6/pkg/frontend/openshiftcluster_list_test.go) and in other places).

Makes it easier to understnd test cases.

Reply:
Added descriptive test names.

Reply:
Added descriptive test names.